### ISSUE TITLE
nullable enums

### ISSUE THREAD

**Issue by phyratokar** [springdoc/springdoc-openapi#1889]
Hi

I'm struggling with generating an API doc that has nullable enums in it. Maybe I'm missing something but I think it's a missing feature.

## What am I trying to achieve:
We have an endpoint that accepts a request, this requests contains a field of enum type which is nullable. So imagine something like this:

``` java
class Request {

  @Nullable
  @Schema(nullable = true, example = "RED")
  private Color color;
}

enum Color {
  RED,
  GREEN,
  YELLOW
}
```

Once we use springdoc-openapi to generate the swagger api definitions we get back the following yml:

``` yml
[...]
components:
  schemas:
    Request:
      type: object
      properties:
        color:
          type: string
          nullable: true
          example: RED
          enum:
          - RED
          - GREEN
          - YELLOW # missing entry for null
```
This allows the request to omit the color property, but if anyone sends a color = null, this is not a valid request according to the api, because null is not listed as a value in the enum.

I'm using springdoc-openapi version 1.6.11.

## Request
Because annotations like @NotNull are also picked up, I would expect @Nullable to be picked up as well and therefore null should be added to the list of possible values.

But if that is not possible for some reason, is there a workaround for such a situation? I could not find any way of adding the needed null entry, also allowableValues does not accept null as a value.
Basically I need just a way to add null to the enum values, it does not really matter how I achieve it.

Thanks in advance for any responses, I hope I did not miss anything important.

**Comment by bnasslahsen** [comment]
@phyratokar, 

There is no such a feature.
The workaround to add the null to the enum value is is to fill the `allowableValues` using `@Schema` annotation

**Comment by phyratokar** [comment]
Well adding null to the allowableValues does not work as it throws a "Attribute value must be constant". So can you tell me how to add null as a valid entry to allowableValues? (adding `"null"` also does not work as it adds the string "null" to the list not actually null)

And besides that, I think it is unexpected behavior to have something that is defined as nullable = true, but then null is not allowed. Don't you agree?

**Comment by bnasslahsen** [comment]
@phyratokar,

You have been already answered.
We use GitHub issues to track bugs and enhancements. If you have a general usage question please ask on [Stack Overflow](https://stackoverflow.com/). The [springdoc](https://stackoverflow.com/tags/springdoc)-openapi team and the broader community monitor the springdoc tag.
